e2e/types/config_test.go

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package types_test
+
+import (
+	"testing"
+
+	"github.com/ramendr/ramen/e2e/types"
+)
+
+func baseConfig() *types.Config {
+	return &types.Config{
+		Distro:     "ocp",
+		Repo:       types.RepoConfig{URL: "https://github.com/org/repo", Branch: "main"},
+		DRPolicy:   "dr-policy",
+		ClusterSet: "clusterset",
+		Clusters: map[string]types.ClusterConfig{
+			"hub": {Kubeconfig: "hub-kubeconfig"},
+			"c1":  {Kubeconfig: "c1-kubeconfig"},
+			"c2":  {Kubeconfig: "c2-kubeconfig"},
+		},
+		PVCSpecs: []types.PVCSpecConfig{
+			{Name: "pvc-a", StorageClassName: "standard", AccessModes: "ReadWriteOnce"},
+		},
+		Tests: []types.TestConfig{
+			{Workload: "wl1", Deployer: "ocm-hub", PVCSpec: "pvc-a"},
+		},
+		Channel: types.ChannelConfig{
+			Name:      "my-channel",
+			Namespace: "ramen-system",
+		},
+		Namespaces: types.NamespacesConfig{
+			RamenHubNamespace:       "ramen-hub",
+			RamenDRClusterNamespace: "ramen-dr-cluster",
+			RamenOpsNamespace:       "ramen-ops",
+			ArgocdNamespace:         "argocd",
+		},
+	}
+}
+
+func TestConfigEqual(t *testing.T) {
+	type test struct {
+		Name        string
+		Modify      func(c *types.Config)
+		ExpectEqual bool
```

Comment:
We can simplify by including only the non equal tests in the list.

We can have different test for equal configs:

- comparing to itself
- comparing to other config with same content

Both can be handle by trivial test:

```go
func TestConfigEqual(t *testing.T) {
    c1 := baseConfig()
    c2 := baseConfig()

    t.Run("to itself", func(t *testing.T) {
        if !c1.Equal(c1) {
            ...
        } 
    }
    t.Run("to other", func(t *testing.T) {
        if !c1.Equal(c2) {
            ...
        } 
    }
}

func TestConfigNotEqual(t *testing.T) {
   ...
}
```